### PR TITLE
W-16640190: Avoid deadlock when consuming the payload in the whenComplete callback of sendAsync result

### PR DESCRIPTION
**Pull request by eze210**

### CODE REVIEW

**Comment by eze210** [comment]
--validate

File: src/test/java/org/mule/service/http/impl/service/client/async/ResponseBodyDeferringAsyncHandlerTestCase.java [hunk 1]
```diff
@@ -269,16 +289,58 @@ public void abortsWhenPipeIsClosed() throws Exception {
...
+    when(headersMap.getFirstValue(CONTENT_LENGTH)).thenReturn(Long.toString((long) Integer.MAX_VALUE * 2));
+    when(headersMap.getFirstValue(TRANSFER_ENCODING)).thenReturn("");
 
     HttpResponseHeaders headers = mock(HttpResponseHeaders.class);
     when(headers.getHeaders()).thenReturn(headersMap);
 
     assertThat(handler.onHeadersReceived(headers), is(CONTINUE));
   }
+
+  @Test
```
Comment (elrodro83) [inline src/test/java/org/mule/service/http/impl/service/client/async/ResponseBodyDeferringAsyncHandlerTestCase.java:305]: `@Issue`

File: src/test/java/org/mule/service/http/impl/service/client/async/ResponseBodyDeferringAsyncHandlerTestCase.java [hunk 2]
```diff
@@ -269,16 +289,58 @@ public void abortsWhenPipeIsClosed() throws Exception {
...
+    });
+
+    assertThat(handler.onBodyPartReceived(intermediatePart), is(CONTINUE));
+    assertThat(handler.onBodyPartReceived(lastPart), is(CONTINUE));
+    assertThat(handler.onCompleted(), is(nullValue()));
+
+    prober.check(new JUnitProbe() {
+
+      @Override
+      protected boolean test() throws Exception {
```
Comment (elrodro83) [inline src/test/java/org/mule/service/http/impl/service/client/async/ResponseBodyDeferringAsyncHandlerTestCase.java:327]: use a lambda

File: src/test/java/org/mule/service/http/impl/service/client/async/ResponseBodyDeferringAsyncHandlerTestCase.java [hunk 3]
```diff
@@ -269,16 +289,58 @@ public void abortsWhenPipeIsClosed() throws Exception {
...
+        assertThat(responseContent.get(), not(nullValue()));
```
Comment (elrodro83) [inline src/test/java/org/mule/service/http/impl/service/client/async/ResponseBodyDeferringAsyncHandlerTestCase.java:328]: not needed based on the following assertion

**Review by elrodro83 — DISMISSED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**